src/atomicskiplist.ml

```diff
@@ -0,0 +1,221 @@
+type 'a markable_reference = { node : 'a; marked : bool }
+(** markable reference: stores a reference to a node and has a field to specify if it is marked *)
+
+type 'a node = {
+  key : int;
+  height : int;
+  next : 'a node option markable_reference Atomic.t array;
+}
+
+type 'a t = { head : 'a node }
+
+let max_height = 10
+
+(** create_new_node: creates a new node with some value and height *)
+let create_new_node value height =
+  let next = Array.init (max_height + 1) (fun _ -> Atomic.make {node = None; marked = false;})
+  in
+  { key = value; height; next }
+
+(** create_dummy_node_array: Creates a new array with the different node for each index *)
+let create_dummy_node_array () =
+  let arr = Array.init (max_height + 1) (fun _ -> create_new_node Int.max_int max_height)
+  in
+  arr
+
+(** Get a random level from 1 till max_height (both included) *)
+let get_random_level () =
+  let rec count_level cur_level =
+    if cur_level == max_height || Random.float 1.0 <= 0.5 then cur_level
```

Comment:
Why not use `Random.bool ()`?

Reply:
BTW, instead of using a loop, one could roughly:
1. Compute `max_height - 1` random bits: `Random.bits () land ((1 lsl (max_height - 1)) - 1) lor (1 lsl (max_height - 1))`.
2. Use the technique described in [this paper](http://supertech.csail.mit.edu/papers/debruijn.pdf) to find the index of the lowest 1 bit.
3. Add 1.